core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -305,6 +305,32 @@ public static boolean notContainsCorrelation(RelNode r,
     }
   }
 
+  /** Rewrites an {@link RexNode} on top of a correlated join to the rights context by
+   * shifting {@link RexInputRef} if they are from the right side, and rewriting
+   * {@link RexInputRef} to field accesses of {@link RexCorrelVariable}.
+   * @param left Rel fields to be correlated
+   * @param id id to correlate
+   * @param right Rel field to be shifted to the right
+   * @param rexNode expression to be rewritten
+   * @return An expression
+   */
```

Comment:
Suggestion below:
```
Transforms a join condition over two inputs to a correlated condition over the right input.

Replaces references on the left relation with correlating variables and shifts references on the right relation as needed to be able to put the condition over the right relation.

TODO: Add example similar to the one below.
<pre>{@code
Join[cond=..]
  TableScan[EMP]
  TableScan[DEPT]
}</pre>

<pre>{@code
Operator
  TableScan[EMP]
  Filter[cond=..]
    TableScan[DEPT]
}</pre>
```
Rename method to either:
```
transformJoinConditionToCorrelate(RexNode condition, RelNode left, RelNode right, CorrelationId id, RexBuilder builder)
transformJoinToCorrelateCondition(...)
transformConditionFromJoinToCorrelate(...)
```

Reply:
@zabetak, I have followed up on your requests: adding an example and renaming the method.  Please let me know if you want any further changes.